src/movies.js

```diff
@@ -1,16 +1,56 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
-
+console.log(movies.length);
+const getAllDirectors = (array) => {
+    const directores = array.map((e) => {
+        return e.director
+    })
+    return directores;
+}
+getAllDirectors(movies)
+console.log(getAllDirectors(movies));
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+const howManyMovies = (arr) => ((arr.filter(e => e.director === "Steven Spielberg")).filter(e => e.genre.includes("Drama"))).length;
+console.log(howManyMovies(movies));
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
-
+const ratesAverage = (array) => {
+    const promedios = array.reduce((acumulado, actual) => {
+        if (actual.rate !== null && actual.rate !== '' && typeof actual.rate !== "undefined") {
+            acumulado += actual.rate;
+        }
+        return acumulado;
+    }, 0)
+    if (promedios === 0) {
+        return 0
+    }
+    let puntuacion = (promedios / array.length)
+    return parseInt(puntuacion.toFixed(2))
```

Comment:
Hola, ojo aquí, no cerraste el bloque de código de tu función howManyMovies
```suggestion
    return parseInt(puntuacion.toFixed(2))
    }
```